app/models/assignment.rb

```diff
@@ -379,42 +379,125 @@ def clone_groupings_from(assignment_id)
+          if grouping_for_current_assignment.nil?
+            if same_membership_as_csv_row?(row, any_grouping)
+              # CASE: Group already exists with the same repo name and has a
+              #     grouping for another assignment with the same membership
+              # Use existing group and create a new grouping between the
+              # existing  group and the given students and return without error
+              add_new_grouping_for_group(row, group)
+              return
+            else
+              # CASE: Group already exists with the same repo name and has
+              #     a grouping for another assignment BUT with different
+              #     membership
+              # The existing groupings and the current group is not compatible
+              # Return an error.
+              duplicate_group_error = I18n.t(
+                "csv.group_with_different_membership_different_assignment",
+                group_name: row[0])
+              return duplicate_group_error
+            end
+          else
+            if same_membership_as_csv_row?(row,
+                                           grouping_for_current_assignment)
+              # CASE: Group already exists with the same repo name and also has
+              #     a grouping for the current assignment with the same
+              #     membership
+              # No new group or grouping created. Since the exact group given by
+              # the csv file already exists treat this as a successful case
+              # and don't return an error
+              return
+            else
+              # CASE: Group already exists with the same repo name and has a
+              #     grouping for the current assignment BUT the membership is
+              #     different.
+              # Return error since the membership is different
+              duplicate_group_error = I18n.t(
+                "csv.group_with_different_membership_current_assignment",
+                group_name: row[0])
+              return duplicate_group_error
+            end
+          end
+        end
+      end
+
     end
 
-    # Since repo_name of "group" will be set before the first save call, the
-    # set repo_name will be used instead of the autogenerated name. See
-    # set_repo_name and build_repository in the groups model. Also, see
-    # create_group_for_working_alone_student in the students model for
-    # similar semantics.
-    if is_candidate_for_setting_custom_repo_name?(row)
-      # Do this only if user_name exists and is a student.
-      if Student.where(user_name: row[2]).first
-        group.repo_name = row[0]
+    # If any of the given members do not exist or is part of another group,
+    # an error is returned without creating a group
+    unless membership_unique?(row)
+      if !errors[:groupings].blank?
+        # groupings error set if a member is already in different group
+        membership_error = I18n.t("csv.memberships_not_unique",
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.